platform/security/rest/security-rest-clientapi/src/main/java/org/codice/ddf/cxf/client/ClientBuilder.java

```diff
@@ -37,7 +37,7 @@
    * @param endpointUrl
    * @return
    */
-  ClientBuilder<T> endpoint(URI endpointUrl);
+  ClientBuilder<T> endpoint(String endpointUrl);
```

Comment:
So why does this need to be String vs URI? I tried to make it a URI to make the API a little tighter. Moving to String basically opens it wide. Would it be possible to keep the API as a URI and instead do uri.toString() in order to resolve the properties and then re set the URI?

Reply:
When we create a source, if we leave the URL with system properties in it, like:
`${org.codice.ddf.external.protocol}${org.codice.ddf.external.hostname}:${org.codice.ddf.external.port}${org.codice.ddf.external.context}${org.codice.ddf.system.rootContext}/csw`
DDF throws an exception because we try to change it to a URI.

The property resolver is in the client builder/factory so unless we move the property resolver to everyone that makes a call like that, we need to change the endpoint to be a String and do the property resolving step in the client builder.